fix(ci): fix `check-node.yml` matrix incompatibility with `engine-strict=true`

# fix(ci): fix `check-node.yml` matrix incompatibility with `engine-strict=true`

| ⏱️ Estimate | 📊 Priority | 📏 Size | 📅 Start | 📅 End |
| --- | --- | --- | --- | --- |
| 1h | P0 | S | 12-04-2026 | 01-05-2026 |

## 📸 Screenshots

| Current | Expected |
| :---: | :---: |
| <img width="800" height="auto" alt="Image" src="https://github.com/user-attachments/assets/31cbc517-e646-4308-849f-a8ac4fa1d53e" /> <img width="800" height="auto" alt="Image" src="https://github.com/user-attachments/assets/ebc27de8-af37-48d9-bd92-205d44326490" /> | N/A — This change has no visual impact. |

## 📝 Summary

- Remove `.noderc.json` to allow unrestricted LTS updates
- Add `deploy.yml` to `update-node-npm.yml` automated workflow
- Fix `check-node.yml` matrix from `[20, 22]` to `[22.22.1]`
- Update `update-node-npm.yml` to propagate exact versions

## 💡 Why this bug?

> [!CAUTION]
> ```bash
> Run npm install
> npm error code EBADENGINE
> npm error engine Unsupported engine
> npm error engine Not compatible with your version of node/npm: vue-gh-pages@3.6.0
> npm error notsup Not compatible with your version of node/npm: vue-gh-pages@3.6.0
> npm error notsup Required: {"node":"22.22.1","npm":"10.9.4"}
> npm error notsup Actual:   {"npm":"10.8.2","node":"v20.20.0"}
> npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2026-03-12T06_45_35_688Z-debug-0.log
> Error: Process completed with exit code 1.
> ```

- The `update-node-npm.yml` workflow cannot update to newer LTS versions because `.noderc.json` sets `maxMajorVersion` to `22`, restricting updates to that major version
- The `deploy.yml` was never included in the `update-node-npm.yml` automated workflow, leaving its `node-version` outdated at `20.x`
- `check-node.yml` tests `[20, 22]` causing #978 to fail CI with `EBADENGINE` errors on Node 20. Reducing to `[22]` still fails because `engine-strict=true` requires exactly `22.22.1`

## ✅ Expected behavior

- `update-node-npm.yml` updates all workflow files (including `deploy.yml`) to the latest LTS version without restrictions
- CI passes with a single Node version matching `engines.node`

## 📋 Steps

### Phase 0: Verify Node 24 compatibility
Before removing `.noderc.json`, verify the project works with Node 24 (the next LTS after 22):

- [x] Switch to Node 24:

```bash
nvm install 24
nvm use 24
```
- [x] Run install:

```bash
npm install
```
- [x] Run build:

```bash
npm run build
```
- [x] Run serve:

```bash
npm run serve
```

### Phase 1: Remove `maxMajorVersion` restriction
- [x] Delete `.noderc.json` so the `update-node-npm.yml` workflow can update to the latest LTS version freely:

```bash
rm .noderc.json
```
```diff
- {
- 	"maxMajorVersion": 22
- }
```

### Phase 2: Add `deploy.yml` support to `update-node-npm.yml`

- [x] Add `deploy.yml` to the header comment in `update-node-npm.yml`:

```diff
 # Files updated by this workflow:
 # - .nvmrc
 # - package.json (engines.node and engines.npm)
 # - .github/workflows/check-node.yml (matrix node-version)
+# - .github/workflows/deploy.yml (node-version)
```
- [x] Add step in `update-node-npm.yml` to also update `node-version` in `deploy.yml`:

```diff
      - name: 📝 Update check-node.yml matrix
        if: steps.check-update.outputs.update_needed == 'true' && steps.current-versions.outputs.has_node_yml == 'true'
        run: |
          ...
+
+     - name: 📝 Update deploy.yml node-version
+       if: steps.check-update.outputs.update_needed == 'true'
+       run: |
+         sed -i "s/node-version: [0-9]*\.x/node-version: ${{ steps.node-versions.outputs.lts_latest }}.x/" .github/workflows/deploy.yml
+         echo "Updated deploy.yml node-version"
+         grep "node-version:" .github/workflows/deploy.yml
+
      - name: 🚀 Create Pull Request
        ...
```
- [x] Add `deploy.yml` reference to the commit message in `update-node-npm.yml`:

```diff
            - Update `check-node.yml` file in matrix: ... → ...
+           - Update `deploy.yml` file in node-version: ...
          title: ...
```
- [x] Add `deploy.yml` reference to the PR body in `update-node-npm.yml`:

```diff
            - Update `check-node.yml` file in matrix: `...` → `...`
+           - Update `deploy.yml` file in node-version: `...`

            ## 🧪 Tests
```
- [x] Add `DEPLOY_YML_ICON` variable and `deploy.yml` row to the Summary step in `update-node-npm.yml`:

```diff
          if [ "${{ steps.current-versions.outputs.has_node_yml }}" == "true" ]; then
            NODE_YML_ICON="✅ Found"
          else
            NODE_YML_ICON="❌ Not found"
          fi
+
+         if [ -f .github/workflows/deploy.yml ]; then
+           DEPLOY_YML_ICON="✅ Found"
+         else
+           DEPLOY_YML_ICON="❌ Not found"
+         fi
```

```diff
          | \`check-node.yml\` | ${NODE_YML_ICON} | CI workflow matrix |
+         | \`deploy.yml\` | ${DEPLOY_YML_ICON} | Deploy workflow node-version |
```

### Phase 3: Update `deploy.yml` node-version
- [x] Update `deploy.yml` from `20.x` to `22.x` to match the current major version:

```diff
      - name: 🛠️ Setup Node version
        uses: actions/setup-node@v6.3.0
        with:
-          node-version: 20.x
+          node-version: 22.x
```

### Phase 4: Address `check-node.yml` matrix issue
With `engine-strict=true` and an exact version in `engines`, only one Node version can pass CI. The original purpose of testing multiple versions is incompatible with strict engine enforcement:

```jsonc
// package.json
"engines": {
  "node": "22.22.1",
  "npm": "10.9.4"
}
```
```yaml
# check-node.yml
matrix:
  node-version: [20, 22]
```

#### `check-node.yml`

- [x] Option A: Remove the matrix and test only the current major version:

```diff
 strategy:
   matrix:
-    node-version: [20, 22]
+    node-version: [22]
```
#### `.npmrc`

- [ ] ~~Option B: Remove `engine-strict=true` from `.npmrc` to allow multiple versions~~ — Discarded

```diff
- engine-strict=true
```

### Phase 5: Pin exact versions
Workflows were using major version (`22`) or wildcard (`22.x`) which resolves to whatever the runner has. With `engine-strict=true`, this causes CI failures when the runner lags behind the exact `engines.node` version. All workflows must use exact versions, and `update-node-npm.yml` must propagate them.

#### `check-node.yml`

- [x] Update matrix from `[22]` to `[22.22.1]`:

```diff
 strategy:
   matrix:
-    node-version: [22]
+    node-version: [22.22.1]
```

#### `deploy.yml`

- [x] Update node-version from `22.x` to `22.22.1`:

```diff
      - name: 🛠️ Setup Node version
        uses: actions/setup-node@v6.3.0
        with:
-          node-version: 22.x
+          node-version: 22.22.1
```

#### `update-node-npm.yml`

- [x] Update the matrix output to generate a single exact version, preventing the workflow from reverting the fix:

```diff
      - name: 🔍 Determine matrix versions
        id: matrix-versions
        run: |
-         echo "matrix=[ $LTS_OLDEST, $LTS_LATEST ]" >> $GITHUB_OUTPUT
+         echo "matrix=[$LATEST_VERSION]" >> $GITHUB_OUTPUT
```
- [x] Update `deploy.yml` sed pattern to match and replace exact versions:

```diff
      - name: 📝 Update deploy.yml node-version
        if: steps.check-update.outputs.update_needed == 'true'
        run: |
-          sed -i "s/node-version: [0-9]*\.x/node-version: ${{ steps.node-versions.outputs.lts_latest }}.x/" .github/workflows/deploy.yml
+          sed -i "s/node-version: [0-9][0-9.]*/node-version: ${{ steps.node-versions.outputs.latest_version }}/" .github/workflows/deploy.yml
           echo "Updated deploy.yml node-version"
           grep "node-version:" .github/workflows/deploy.yml
```
- [x] Update commit message reference from wildcard to exact version:

```diff
            - Update `check-node.yml` file in matrix: ... → ...
-            - Update `deploy.yml` file in node-version: ${{ steps.node-versions.outputs.lts_latest }}.x
+            - Update `deploy.yml` file in node-version: ${{ steps.node-versions.outputs.latest_version }}
          title: ...
```
- [x] Update PR body reference from wildcard to exact version:

```diff
            - Update `check-node.yml` file in matrix: `...` → `...`
-            - Update `deploy.yml` file in node-version: `${{ steps.node-versions.outputs.lts_latest }}.x`
+            - Update `deploy.yml` file in node-version: `${{ steps.node-versions.outputs.latest_version }}`

            ## 🧪 Tests
```

### Phase 6: Close related PR
- [x] Add a comment on PR #978 explaining this issue and close the PR

## 🧪 Tests

- [x] Verify `update-node-npm.yml` workflow runs successfully without `.noderc.json`
- [x] Verify `deploy.yml` is updated when `update-node-npm.yml` runs
- [x] Run install locally:

```bash
npm install
```
- [x] Run build locally:

```bash
npm run build
```
- [x] Run serve locally:

```bash
npm run serve
```

## 📌 Notes

### Version changes

| File | From | To | Type |
| --- | --- | --- | --- |
| `.noderc.json` | `maxMajorVersion: 22` | Deleted | 🔴 Major |
| `.github/workflows/deploy.yml` | `node-version: 20.x` | `node-version: 22.22.1` | 🔴 Major |
| `.github/workflows/check-node.yml` | `matrix: [20, 22]` | `matrix: [22.22.1]` | 🟡 Minor |

- Node 24 was tested locally and works correctly (`npm install`, `npm run build` and `npm run serve`)
- Once `.noderc.json` is removed, the next `update-node-npm.yml` run will update all files to the latest LTS

## 🔗 References

### Files to modify
- `.github/workflows/update-node-npm.yml`
- `.github/workflows/deploy.yml`
- `.github/workflows/check-node.yml`

### Files to remove
- `.noderc.json`

### Related PRs
- #978
- #984

--- a/.github/workflows/check-node.yml
+++ b/.github/workflows/check-node.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [20, 22]
+        node-version: [22]
     steps:
       - name: 🔀 Checkout code from repository
         uses: actions/checkout@v6.0.1

--- a/.github/workflows/check-node.yml
+++ b/.github/workflows/check-node.yml
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [22]
+        node-version: [22.22.1]
     steps:
       - name: 🔀 Checkout code from repository
         uses: actions/checkout@v6.0.1

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -14,7 +14,7 @@ jobs:
       - name: 🛠️ Setup Node version
         uses: actions/setup-node@v6.3.0
         with:
-          node-version: 22.x
+          node-version: 22.22.1
       - name: 📦 Install dependencies
         run: npm install
       - name: 🙍‍♂️ Setup git user

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -14,7 +14,7 @@ jobs:
       - name: 🛠️ Setup Node version
         uses: actions/setup-node@v6.3.0
         with:
-          node-version: 20.x
+          node-version: 22.x
       - name: 📦 Install dependencies
         run: npm install
       - name: 🙍‍♂️ Setup git user

--- a/.github/workflows/update-node-npm.yml
+++ b/.github/workflows/update-node-npm.yml
@@ -156,7 +156,7 @@ jobs:
           echo "npm_version=$NPM_VERSION" >> $GITHUB_OUTPUT
           echo "lts_oldest=$LTS_OLDEST" >> $GITHUB_OUTPUT
           echo "lts_latest=$LTS_LATEST" >> $GITHUB_OUTPUT
-          echo "matrix=[$LTS_LATEST]" >> $GITHUB_OUTPUT
+          echo "matrix=[$LATEST_VERSION]" >> $GITHUB_OUTPUT
           echo "has_version_limit=$HAS_CONFIG" >> $GITHUB_OUTPUT
 
       - name: 📖 Get current versions
@@ -243,7 +243,7 @@ jobs:
       - name: 📝 Update deploy.yml node-version
         if: steps.check-update.outputs.update_needed == 'true'
         run: |
-          sed -i "s/node-version: [0-9]*\.x/node-version: ${{ steps.node-versions.outputs.lts_latest }}.x/" .github/workflows/deploy.yml
+          sed -i "s/node-version: [0-9][0-9.]*/node-version: ${{ steps.node-versions.outputs.latest_version }}/" .github/workflows/deploy.yml
           echo "Updated deploy.yml node-version"
           grep "node-version:" .github/workflows/deploy.yml
 
@@ -266,7 +266,7 @@ jobs:
             - Update `package.json` file in `engines.node`: ${{ steps.current-versions.outputs.current_node }} → ${{ steps.node-versions.outputs.latest_version }}
             - Update `package.json` file in `engines.npm`: ${{ steps.current-versions.outputs.current_npm }} → ${{ steps.node-versions.outputs.npm_version }}
             - Update `check-node.yml` file in matrix: ${{ steps.current-versions.outputs.current_matrix }} → ${{ steps.node-versions.outputs.matrix }}
-            - Update `deploy.yml` file in node-version: ${{ steps.node-versions.outputs.lts_latest }}.x
+            - Update `deploy.yml` file in node-version: ${{ steps.node-versions.outputs.latest_version }}
           title: "build(deps): update `node@${{ steps.node-versions.outputs.latest_version }}` and `npm@${{ steps.node-versions.outputs.npm_version }}` versions"
           body: |
             # build(deps): update `node@${{ steps.node-versions.outputs.latest_version }}` and `npm@${{ steps.node-versions.outputs.npm_version }}` versions
@@ -295,7 +295,7 @@ jobs:
             - Update `package.json` file in `engines.node`: `${{ steps.current-versions.outputs.current_node }}` → `${{ steps.node-versions.outputs.latest_version }}`
             - Update `package.json` file in `engines.npm`: `${{ steps.current-versions.outputs.current_npm }}` → `${{ steps.node-versions.outputs.npm_version }}`
             - Update `check-node.yml` file in matrix: `${{ steps.current-versions.outputs.current_matrix }}` → `${{ steps.node-versions.outputs.matrix }}`
-            - Update `deploy.yml` file in node-version: `${{ steps.node-versions.outputs.lts_latest }}.x`
+            - Update `deploy.yml` file in node-version: `${{ steps.node-versions.outputs.latest_version }}`
 
             ## 🧪 Tests
 

--- a/.github/workflows/update-node-npm.yml
+++ b/.github/workflows/update-node-npm.yml
@@ -13,6 +13,7 @@
 # - .nvmrc
 # - package.json (engines.node and engines.npm)
 # - .github/workflows/check-node.yml (matrix node-version)
+# - .github/workflows/deploy.yml (node-version)
 
 # Optional configuration file (.noderc.json):
 # {
@@ -239,6 +240,13 @@ jobs:
           echo "Updated check-node.yml matrix"
           grep "node-version:" .github/workflows/check-node.yml
 
+      - name: 📝 Update deploy.yml node-version
+        if: steps.check-update.outputs.update_needed == 'true'
+        run: |
+          sed -i "s/node-version: [0-9]*\.x/node-version: ${{ steps.node-versions.outputs.lts_latest }}.x/" .github/workflows/deploy.yml
+          echo "Updated deploy.yml node-version"
+          grep "node-version:" .github/workflows/deploy.yml
+
       - name: 🚀 Create Pull Request
         if: |
           steps.check-update.outputs.update_needed == 'true' &&
@@ -258,6 +266,7 @@ jobs:
             - Update `package.json` file in `engines.node`: ${{ steps.current-versions.outputs.current_node }} → ${{ steps.node-versions.outputs.latest_version }}
             - Update `package.json` file in `engines.npm`: ${{ steps.current-versions.outputs.current_npm }} → ${{ steps.node-versions.outputs.npm_version }}
             - Update `check-node.yml` file in matrix: ${{ steps.current-versions.outputs.current_matrix }} → ${{ steps.node-versions.outputs.matrix }}
+            - Update `deploy.yml` file in node-version: ${{ steps.node-versions.outputs.lts_latest }}.x
           title: "build(deps): update `node@${{ steps.node-versions.outputs.latest_version }}` and `npm@${{ steps.node-versions.outputs.npm_version }}` versions"
           body: |
             # build(deps): update `node@${{ steps.node-versions.outputs.latest_version }}` and `npm@${{ steps.node-versions.outputs.npm_version }}` versions
@@ -286,6 +295,7 @@ jobs:
             - Update `package.json` file in `engines.node`: `${{ steps.current-versions.outputs.current_node }}` → `${{ steps.node-versions.outputs.latest_version }}`
             - Update `package.json` file in `engines.npm`: `${{ steps.current-versions.outputs.current_npm }}` → `${{ steps.node-versions.outputs.npm_version }}`
             - Update `check-node.yml` file in matrix: `${{ steps.current-versions.outputs.current_matrix }}` → `${{ steps.node-versions.outputs.matrix }}`
+            - Update `deploy.yml` file in node-version: `${{ steps.node-versions.outputs.lts_latest }}.x`
 
             ## 🧪 Tests
 
@@ -340,6 +350,12 @@ jobs:
             NODE_YML_ICON="❌ Not found"
           fi
 
+          if [ -f .github/workflows/deploy.yml ]; then
+            DEPLOY_YML_ICON="✅ Found"
+          else
+            DEPLOY_YML_ICON="❌ Not found"
+          fi
+
           cat >> $GITHUB_STEP_SUMMARY << EOF
           # Node.js Update Check Summary
 
@@ -363,6 +379,7 @@ jobs:
           | \`.nvmrc\` | ${NVMRC_ICON} | Node version for nvm |
           | \`package.json\` | ${ENGINES_ICON} | Engines of node and npm |
           | \`check-node.yml\` | ${NODE_YML_ICON} | CI workflow matrix |
+          | \`deploy.yml\` | ${DEPLOY_YML_ICON} | Deploy workflow node-version |
 
           ## 🔗 Resources
 

--- a/.github/workflows/update-node-npm.yml
+++ b/.github/workflows/update-node-npm.yml
@@ -156,7 +156,7 @@ jobs:
           echo "npm_version=$NPM_VERSION" >> $GITHUB_OUTPUT
           echo "lts_oldest=$LTS_OLDEST" >> $GITHUB_OUTPUT
           echo "lts_latest=$LTS_LATEST" >> $GITHUB_OUTPUT
-          echo "matrix=[ $LTS_OLDEST, $LTS_LATEST ]" >> $GITHUB_OUTPUT
+          echo "matrix=[$LTS_LATEST]" >> $GITHUB_OUTPUT
           echo "has_version_limit=$HAS_CONFIG" >> $GITHUB_OUTPUT
 
       - name: 📖 Get current versions

--- a/.noderc.json
+++ b/.noderc.json
@@ -1,3 +1,0 @@
-{
-	"maxMajorVersion": 22
-}

--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
 		"deploy:v2": "node deploy-v2.mjs"
 	},
 	"dependencies": {
-		"chalk": "^5.6.0",
+		"chalk": "^5.6.2",
 		"core-js": "^3.48.0",
 		"execa": "^9.6.1",
 		"node-emoji": "^2.2.0",

--- a/package.json
+++ b/package.json
@@ -83,10 +83,16 @@
 		"npm": "10.9.4"
 	},
 	"overrides": {
-		"stylelint-config-recommended": "17.0.0",
 		"cross-spawn": "^7.0.6",
-		"webpack-dev-server": "^5.2.1",
-		"yorkie": "^2.0.0",
+		"stylelint-config-recommended-vue": {
+			"stylelint-config-recommended": "17.0.0"
+		},
+		"@vue/cli-service": {
+			"webpack-dev-server": "^5.2.1"
+		},
+		"@vue/cli-plugin-eslint": {
+			"yorkie": "^2.0.0"
+		},
 		"@vue/component-compiler-utils": {
 			"postcss": "^8.4.47"
 		},

--- a/package.json
+++ b/package.json
@@ -88,6 +88,7 @@
 			"stylelint-config-recommended": "17.0.0"
 		},
 		"@vue/cli-service": {
+			"serialize-javascript": "^7.0.4",
 			"webpack-dev-server": "^5.2.1"
 		},
 		"@vue/cli-plugin-eslint": {

--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-	"version": "3.5.1",
+	"version": "3.6.0",
 	"private": true,
 	"name": "vue-gh-pages",
 	"description": "Tutorial to publish a public repository made with Vue in GithubPages.",

